lib/elixir/lib/module.ex

```diff
@@ -60,11 +60,14 @@ defmodule Module do
   To aid in the correct implementation of behaviours, you may optionally declare
   `@impl` for implemented callbacks of a behaviour. This makes callbacks
   explicit and can help you to catch errors in your code. The compiler will warn
-  you if you mark a function as `@impl` when in fact it is not a callback, and
-  vice-versa. It also helps with maintainability by making it clear to other
-  developers that the function's purpose is to implement a callback.
+  you if you mark a function as `@impl` when in fact it is not a callback. It will
+  also warn when you mark a function with `@impl` when it isn't a callback, but only
+  if you marked other callbacks with `@impl`.
 
```

Comment:
What do you think about clarifying that:

* if you mark a function with `@impl`, you must mark all other functions of that behavior with `@impl`. Otherwise, you get: 
  ```
  warning: module attribute @impl was not set for function bar/0 callback (specified in Foo). 
  This either means you forgot to add the "@impl true" annotation before the definition or that you are accidentally overriding this callback
  ```
* `@impl` is applied per context. If you mark functions generated by a macro with `@impl`, that won't affect the functions in the module you `use/2` that macro.

Reply:
Good call. I rephrased your suggestion but added it in the latest commit.